EIPS/eip-5773.md

```diff
@@ -0,0 +1,362 @@
+
+Such NFT can be given further utility by means of new additional resources: more games, more cosmetic items, appended to the same NFT. Thus, a game cosmetic item as an NFT becomes an ever-evolving NFT of infinite utility.
+
+### Resource fields
+
+The MultiResource token standard supports two resource fields:
+
+- `id`: a `uint64` resource identifier
+- `metadataURI`: a `string` pointing to the metadata URI associated with the resource
+
+### Multi-Resource Storage Schema
+
+Resources are stored within a token as an array of `uint64` identifiers.
+
+In order to reduce redundant on-chain string storage, multi resource tokens store resources by reference via inner storage. A resource entry on the storage is stored via a `uint64` mapping to resource data.
+
+A resource array is an array of these `uint64` resource ID references.
+
+Such a structure allows that, a generic resource can be added to the storage one time, and a reference to it can be added to the token contract as many times as we desire. Implementers can then use string concatenation to procedurally generate a link to a content-addressed archive based on the base *SRC* in the resource and the *token ID*. Storing the resource in a new token will only take 16 bytes of storage in the resource array per token for recurrent as well as `tokenId` dependent resources.
+
+Structuring token's resources in such a way allows for URIs to be derived programmatically through concatenation, especially when they differ only by `tokenId`.
+
+### Propose-Commit pattern for resource addition
+
+Adding resources to an existing token MUST be done in the form of a propose-commit pattern to allow for limited mutability by a 3rd party. When adding a resource to a token, it is first placed in the *"Pending"* array, and MUST be migrated to the *"Active"* array by the token's owner. The *"Pending"* resources array SHOULD be limited to 128 slots to prevent spam and griefing.
+
+### Resource management
+
+Several functions for resource management are included. In addition to permissioned migration from "Pending" to "Active", the owner of a token MAY also drop resources from both the active and the pending array -- an emergency function to clear all entries from the pending array MUST also be included.
+
+## Backwards Compatibility
+
+The MultiResource token standard has been made compatible with [EIP-721](./eip-721.md) in order to take advantage of the robust tooling available for implementations of EIP-721 and to ensure compatibility with existing EIP-721 infrastructure.
+
+## Test Cases
+
+The RMRK MultiResource lego block implementation includes test cases written using Hardhat.
+
+## Reference Implementation
+
+RMRK MultiResource lego block and documentation
+
+- Compatible with the original version of the standard
+
+Neon Crisis, by CicadaNCR
+
+- A NFT game utilizing RMRK MultiResource lego block
+
+Snake Soldiers, by Steven Pineda
+
+- A NFT game utilizing RMRK MultiResource lego block
```

Comment:
Please include the reference implementation in the `assets` directory if you have a compatible license (free to redistribute and doesn't place additional restrictions on downstream projects like Apache, MIT, or preferably CC0.)

If you cannot include them, just remove these projects.